src/components/model/ModelForm.svelte

```diff
@@ -0,0 +1,167 @@
+  export let modelId: number | undefined;
+  export let createdAt: string | undefined;
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher<{
+    createPlan: number;
+    deleteModel: void;
+    hasModelChanged: {
+      description: string;
+      name: string;
+      owner: UserId;
+      version: string;
+    };
+  }>();
+
+  const updateModelPermissionError: string = 'You do not have permission to update this model';
+  const deleteModelPermissionError: string = 'You do not have permission to delete this model';
+  const createPlanPermissionError: string = 'You do not have permission to create a plan';
+
+  let hasUpdateModelPermission: boolean = false;
+  let hasDeleteModelPermission: boolean = false;
+  let hasCreatePlanPermission: boolean = false;
+  let name: string = '';
+  let owner: UserId | null = null;
+  let version: string = '';
+  let description: string = '';
+
+  $: description = initialModelDescription;
+  $: name = initialModelName;
+  $: owner = initialModelOwner ?? null;
+  $: version = initialModelVersion ?? '';
+
+  $: if (user) {
+    hasUpdateModelPermission = featurePermissions.model.canUpdate(user);
+    hasDeleteModelPermission = featurePermissions.model.canDelete(user);
+    hasCreatePlanPermission = featurePermissions.plan.canCreate(user);
+  }
+
+  $: dispatch('hasModelChanged', {
+    description,
+    name,
+    owner,
+    version,
+  });
+
+  function onCreatePlanWithModel() {
+    if (modelId !== undefined) {
+      dispatch('createPlan', modelId);
+    }
+  }
+
+  function onDeleteModel() {
+    dispatch('deleteModel');
+  }
+
+  onDestroy(() => {
+    resetModelStores();
+  });
+</script>
+
+<div class="model-form-container">
+  <Input layout="inline">
+    <label for="name">Model name</label>
+    <input
+      class="st-input w-100"
+      name="name"
+      bind:value={name}
+      use:permissionHandler={{
+        hasPermission: hasUpdateModelPermission,
+        permissionError: updateModelPermissionError,
+      }}
+    />
+  </Input>
+  <Input layout="inline">
+    <label for="id">Model id</label>
+    <input class="st-input w-100" disabled name="id" value={modelId ?? ''} />
+  </Input>
+  <Input layout="inline">
+    <label for="description">Model description</label>
+    <textarea
+      class="st-input w-100"
+      name="description"
+      bind:value={description}
+      use:permissionHandler={{
+        hasPermission: hasUpdateModelPermission,
+        permissionError: updateModelPermissionError,
+      }}
+    />
+  </Input>
+  <Input layout="inline">
+    <label for="version">Model version</label>
+    <input
+      class="st-input w-100"
+      name="version"
+      placeholder="0.0.0"
+      bind:value={version}
+      use:permissionHandler={{
+        hasPermission: hasUpdateModelPermission,
+        permissionError: updateModelPermissionError,
+      }}
+    />
+  </Input>
+  <Input layout="inline">
+    <label for="owner">Owner</label>
+    <input
```

Comment:
Future PR for refactoring plan collaborator input and having a user tag input here?

Reply:
Yeah! I actually already have a branch that pulls your plan collaborator input into a hopefully reusable component. I'll open that PR after this gets merged.